xpk.py

```diff
@@ -3449,23 +3403,55 @@ def add_image_pull_policy_for_pw(args):
       YAML stating that the image will be pulled fro GCR every time.
   """
   yaml="""imagePullPolicy: Always"""
-  if args.use_pathways:
+
+  if args.use_pathways or system.accelerator_type == AcceleratorType['GPU']:
     return yaml.format(args=args)
   return ""
 
-def get_pw_volume_mounts(args) -> str:
+def get_main_container_docker_image(args, system: SystemCharacteristics) -> str:
+  """ Docker name for the main container.
+  Args:
+    args: user provided args.
+    system: system characteristics.
+
+  Returns:
+    str:
+      Workload docker image as a YAML string
+  """
+  # Resources requirements for Pathways workload containers are known.
+
+  if system.accelerator_type == AcceleratorType['GPU']:
+    return "maxtext-tcpx"
```

Comment:
Hmm I think the name is just an identifier for the docker image. We can set this to `gpu-image` instead of `maxtext-tcpx` which might not be correct if the user is not running maxtext.

---

Outside of this PR, xpk should support the args.docker_image arg but set a reasonable default for the gpu / cpu /tpu cases. Right now the default is `jax-tpu` which is not reasonable default for gpus or cpus. I can look at this later.